### ISSUE TITLE
Always select closest word

### ISSUE THREAD

**Issue by ProgerXP** [ProgerXP/Notepad2e#205]
This is likely a Scintilla feature.

Document with two lines (`|` indicate cursor positions):

```
|  aa bb
|  'aa bb
```

Place cursor at line 1, Ctrl+Space - `aa` will be selected. This is good.

Place cursor at line 2, Ctrl+Space - the entire line will be selected. This is bad - it should behave as line 1, i.e. locate the closest word by right-scanning and then select it. For selecting a line there is Ctrl+Shift+Space hotkey.

In other words, for both lines selection should be the same.

**Comment by ProgerXP** [comment]
Sorry, I entirely missed the modifier - it's a hotkey, thus it's Ctrl+Space for "select word" (and Ctrl+Shift+Space for "select line").

**Comment by cshnik** [comment]
Fixed.

**Comment by ProgerXP** [comment]
Extend this behaviour so that it works left-side too. Document:

```
aa bb ' |
```

Ctrl+Space currently selects the entire line. Instead, make it select closest left-side word (`bb`), or if there's none - do nothing. In other words, Ctrl+Space will never select a complete line after this change.

**Comment by cshnik** [comment]
Fixed.

**Comment by ProgerXP** [comment]
Document: `¶`

Press Ctrl+Space - the program hangs.

**Comment by cshnik** [comment]
Fixed.

**Comment by ProgerXP** [comment]
48300c9 introduced a subtile change: Ctrl+Space no longer selects word on the next line. Consider this document:

```
first
|
third
```

Pressing Ctrl+Space selected `third` before that commit, now it selects nothing. This was useful; restore it by doing a right-scan across multiple subsequent lines if left-scan within the same line didn't find any word. Use some safety threshold after which scans stop (like 10000 bytes/characters).

**Comment by cshnik** [comment]
Fixed.

**Comment by ProgerXP** [comment]
Must skip non-word symbols:

```
|
// foo
```

Result:

```

// [foo]
```

**Comment by cshnik** [comment]
Fixed.

**Comment by ProgerXP** [comment]
Memo: as it was coded, in absence of any word on the left (on the same line) and on the right until EOF it moves caret to before the first column on the last non-empty line:

```
.|
.

.          <- to here: |.

```

**Comment by ProgerXP** [comment]
> Place cursor at line 2, Ctrl+Space - the entire line will be selected. This is bad - it should behave as line 1, i.e. locate the closest word by right-scanning and then select it. For selecting a line there is Ctrl+Shift+Space hotkey.

As it was implemented by 48300c9ef8818a04869c1deb5807514eb188f4b5 https://github.com/ProgerXP/Notepad2e/issues/205#issuecomment-699536336, when there is a selection Ctrl+Space does nothing (keeps that selection). Adjust it so that it pretends selection is empty and selects the closest word near the cursor (i.e. start or end of the selection depending on caret location). If there's no word, selection is unchanged (as currently), else the word is selected (which may be the same selection or intersect with it).

For example: `a[a a|]a` - Ctrl+Space: `aa [aa|]` - Ctrl+Space: `aa [aa|]` (same).

Another: `a[|a a]a` or `a|a aa` or `|aa aa` or `aa| aa` - Ctrl+Space (and subsequent): `[aa|] aa`.

One case when this is more useful than the current behaviour is doing a Find on a word's part in order to change it (e.g. change case) - currently you have to press arrow left or right to clear the located selection, then Ctrl+Space.

**Comment by cshnik** [comment]
Done.

**Comment by ProgerXP** [comment]
```
[|
aa]
```

Currently, pressing Ctrl+Space clears the selection. Pressing it again selects `[aa]`. It should select that on the first keystroke.

**Comment by cshnik** [comment]
Fixed.
